lib/internal/http2/core.js

```diff
@@ -3380,8 +3380,13 @@ function connect(authority, options, listener) {
 ObjectDefineProperty(connect, promisify.custom, {
   __proto__: null,
   value: (authority, options) => {
-    return new Promise((resolve) => {
-      const server = connect(authority, options, () => resolve(server));
+    return new Promise((resolve, reject) => {
+      const server = connect(authority, options, () => {
+        server.removeListener('error', reject);
```

Comment:
```suggestion
        server.removeListener('error', reject);
```
Why remove the listener on error?

Reply:
That was my suggestion - otherwise errors _after_ connect can be silently swallowed by the leftover error listener here.